src/api/spec/models/workflow/step/submit_request_step_spec.rb

```diff
@@ -0,0 +1,172 @@
+require 'rails_helper'
+
+RSpec.describe Workflow::Step::SubmitRequest, vcr: true do
+  let!(:user) { create(:confirmed_user, :with_home, login: 'Iggy') }
+  let(:other_user) { create(:confirmed_user, :with_home, login: 'Foo') }
+  let(:token) { create(:workflow_token, executor: user) }
+  let(:source_maintainer) { user }
+  let!(:project) { create(:project, name: 'foo_project', maintainer: source_maintainer) }
+  let!(:package) { create(:package_with_file, name: 'bar_package', project: project) }
+  let(:target_project) { create(:project, name: 'baz_project') }
+  let(:scm_webhook) do
+    SCMWebhook.new(payload: {
+                     scm: 'github',
+                     event: 'pull_request',
+                     action: action,
+                     pr_number: 1,
+                     source_repository_full_name: 'reponame',
+                     commit_sha: '123456789',
+                     target_repository_full_name: 'openSUSE/open-build-service'
+                   })
+  end
+  let(:step_instructions) do
+    {
+      source_project: package.project.name,
+      source_package: package.name,
+      target_project: target_project.name
+    }
+  end
+  let(:workflow_run) { create(:workflow_run, token: token) }
+
+  subject do
+    described_class.new(step_instructions: step_instructions,
+                        scm_webhook: scm_webhook,
+                        token: token,
+                        workflow_run: workflow_run)
+  end
+
+  describe '#call' do
+    before do
+      login(user)
+      allow(Backend::Api::Sources::Package).to receive(:wait_service).and_return(true)
+    end
+
+    context 'for a newly opened PR' do
+      let(:action) { 'opened' }
+
+      it 'creates a submit request' do
+        expect { subject.call }.to(change(BsRequest.where(state: 'new'), :count).by(1))
+      end
+
+      it 'creates a event subcription and the workflow artifacts' do
+        expect { subject.call }.to(change(EventSubscription.where(eventtype: 'Event::RequestStatechange'), :count).by(1))
+        expect { subject.call }.to(change(WorkflowArtifactsPerStep.where(workflow_run_id: workflow_run.id), :count).by(1))
+      end
+    end
+
+    context 'for a closed PR' do
+      let(:action) { 'closed' }
+
+      context 'when the token user is authorized' do
+        let!(:bs_request) do
+          create(:bs_request_with_submit_action, source_project: project, source_package: package,
+                                                 target_project: target_project,
+                                                 target_package: package.name, creator: user)
+        end
+
+        it 'revokes previous submit requests' do
+          expect do
+            subject.call
+            bs_request.reload
+          end.to(change(bs_request, :state).from(:new).to(:revoked))
+          expect { subject.call }.not_to change(BsRequest, :count)
+        end
+      end
+
+      context 'when the token user is not authorized' do
```

Comment:
Do not test authorization outside a policy spec please. There is no custom logic involved.